### PR TITLE
sensors: convert lis2dh to use information coming from dts

### PR DESCRIPTION
**Pull request by avisconti**
Convert lis2dh accelerometer driver to get the device name as well
as i2c/spi slave information from device tree. Updates the build_all
test accordingly. (issue #11605)

Signed-off-by: Armando Visconti <armando.visconti@st.com>

### CODE REVIEW

**Comment by avisconti** [comment]
This PR is addressing issue #11605.
I verified building driver for both I2C and SPI, but I tested only I2C on zephyr board.

**Review by galak — APPROVED** [review]

File: dts/bindings/sensor/st,lis2dh.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+description: >
+    This binding gives a base representation of LIS2DH 3-axis accelerometer
+
+inherits:
+    !include i2c-device.yaml
+
+properties:
+    compatible:
+      constraint: "st,lis2dh"
```
Comment (MaureenHelm) [inline dts/bindings/sensor/st,lis2dh.yaml]: @galak, should this be "st,lis2dh-i2c"?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
@avisconti Thanks for your help converting this sensor driver to dts. Can you help with other sensors too? See #9962

File: dts/bindings/sensor/st,lis2dh.yaml (hunk 1 above)
Comment (avisconti) [reply dts/bindings/sensor/st,lis2dh.yaml]: @MaureenHelm 
> should this be "st,lis2dh-i2c"

My opinion is that it does make sense indeed, but if this is a general rule that has to be followed there are other sensors (e.g. lsm6dsl) that need to be changed too.

File: drivers/sensor/lis2dh/Kconfig [hunk 2]
```diff
@@ -8,84 +8,29 @@
...
-	depends on LIS2DH_BUS_I2C
-	default "I2C_0"
-	help
-	  Specify the device name of the I2C master device to which LIS2DH is
-	  connected.
-
-choice
+choice LIS2DH_TRIGGER_MODE
 	prompt "Trigger mode"
-	default LIS2DH_TRIGGER_GLOBAL_THREAD
```
[the remark this answers is not in the document]
Comment (avisconti) [reply drivers/sensor/lis2dh/Kconfig]: ack

**Comment by avisconti** [comment]
> Thanks for your help converting this sensor driver to dts. 
> Can you help with other sensors too? See #9962

Sure, I can take the ST ones

File: drivers/sensor/lis2dh/Kconfig (hunk 2 above)
[the remark this answers is not in the document]
Comment (avisconti) [reply drivers/sensor/lis2dh/Kconfig]: I think that there are also the BUS_TYPE changes not related to the dts. All these changes (i.e. BUS_TYPE and TRIGGER_MODE choices) are useful for configuration selection in board though, so I'll keep them but in a separate commit.

**Comment by avisconti** [comment]
@galak 
I didn't change the yaml for i2c in this re-push. Let us know your opinion and I'll change accordingly if the case.

**Comment by avisconti** [comment]
recheck

**Comment by MaureenHelm** [comment]
Looks good, but would like to get feedback from @galak regarding the i2c compatible.

**Comment by avisconti** [comment]
@galak @MaureenHelm 
If you both agree I would change the i2c comaptible string into "st,lis2dh-i2c".
It make sense and it is a new creation. We may align the others later on.

**Comment by avisconti** [comment]
done!
see if it is ok now.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**